src/Illuminate/Auth/Notifications/ResetPassword.php

```diff
@@ -47,7 +47,7 @@ public function via($notifiable)
     /**
      * Build the mail representation of the notification.
      *
-     * @param  mixed  $notifiable
+     * @param  \Illuminate\Contracts\Auth\CanResetPassword  $notifiable
```

Comment:
This doesn't makes sense because you expect a `notifiable` here and not a `CanResetPassword` instance. You can't be 100% sure that it's a `User` instance you're receiving here. The change below is ok for me but let's keep the `mixed` type-hint.

Reply:
That's fair, I was just lining up with the aforementioned PR. I'll fix this now.

Reply:
Why would a non-`CanResetPassword` implementation be trying to, well, reset their password?

Reply:
There's, for example, the `if (static::$toMailCallback) {` at the beginning which could let you do whatever you want with whatever implementation you have.